src/sentry/grouping/fingerprinting/__init__.py

```diff
@@ -470,17 +381,112 @@ def text(self):
+            new_values = rule.get_fingerprint_values_for_event_access(access)
+            if new_values is not None:
+                return (rule,) + new_values
+        return None
+
+    @classmethod
+    def _from_config_structure(cls: Any, data: dict[str, Any], bases: Any = None) -> Any:
+        version = data["version"]
+        if version != VERSION:
+            raise ValueError("Unknown version")
+        return cls(
+            rules=[Rule._from_config_structure(x) for x in data["rules"]],
+            version=version,
+            bases=bases,
+        )
+
+    def _to_config_structure(self, include_builtin: bool = False) -> None:
+        rules = self.iter_rules(include_builtin=include_builtin)
+
+        return {"version": self.version, "rules": [x._to_config_structure() for x in rules]}
+
+    def to_json(self, include_builtin: bool = False) -> None:
+        return self._to_config_structure(include_builtin=include_builtin)
+
+    @classmethod
+    def from_json(cls, value: dict[str, Any], bases: Any = None) -> Any:
+        try:
+            return cls._from_config_structure(value, bases=bases)
+        except (LookupError, AttributeError, TypeError, ValueError) as e:
+            raise ValueError("invalid fingerprinting config: %s" % e)
+
+    @staticmethod
+    def from_config_string(s: Any, bases: Any = None) -> Any:
+        try:
+            tree = fingerprinting_grammar.parse(s)
+        except ParseError as e:
+            context = e.text[e.pos : e.pos + 33]
+            if len(context) == 33:
+                context = context[:-1] + "..."
+            raise InvalidFingerprintingConfig(
+                f'Invalid syntax near "{context}" (line {e.line()}, column {e.column()})'
+            )
+        return FingerprintingVisitor(bases=bases).visit(tree)
+
+
+class BuiltInFingerprintingRules(FingerprintingRules):
+    """
+    A FingerprintingRules object that marks all of its rules as built-in
+    """
+
+    @staticmethod
+    def from_config_string(s: Any, bases: Any = None) -> None:
+        fingerprinting_rules = FingerprintingRules.from_config_string(s, bases=bases)
+        for r in fingerprinting_rules.rules:
+            r.is_builtin = True
+        return fingerprinting_rules
+
+    @classmethod
+    def _from_config_structure(
+        cls: Any, data: dict[str, Any], bases: Any = None
+    ) -> Sequence[FingerprintingRules]:
+        fingerprinting_rules = super()._from_config_structure(data, bases=bases)
+        for r in fingerprinting_rules.rules:
+            r.is_builtin = True
+        return fingerprinting_rules
+
+
+class FingerprintingVisitor(NodeVisitor):  # type: ignore[type-arg]
     visit_empty = lambda *a: None
     unwrapped_exceptions = (InvalidFingerprintingConfig,)
 
-    def __init__(self, bases):
+    def __init__(self, bases: Any) -> None:
         self.bases = bases
 
-    def visit_comment(self, node, children):
+    def visit_comment(self, node: Any, _: Sequence[Any]) -> str:
```

Comment:
I'm using `Any` for `node` due to the same reason listed above.

Reply:
hmmm I think you can fix it given my suggestion above